glm/ext/quaternion_trigonometric.inl

```diff
@@ -1,9 +1,18 @@
+#include "scalar_constants.hpp"
+
 namespace glm
 {
 	template<typename T, qualifier Q>
 	GLM_FUNC_QUALIFIER T angle(qua<T, Q> const& x)
 	{
-		return acos(x.w) * static_cast<T>(2);
+		if (abs(x.w) > cos_one_over_two<T>())
+		{
+			return asin(sqrt(x.x * x.x + x.y * x.y + x.z * x.z)) * static_cast<T>(2);
+		}
+		else
```

Comment:
Nice, this else can be still removed though due to the return in the if branch

Reply:
Done.